### PR TITLE
signup route (check code)

### PR DESCRIPTION
**Pull request by MohammadKoolivand**
second part of signup added. It contains checking code

### CODE REVIEW

File: src/controllers/app/userController.js [hunk 1]
```diff
@@ -51,3 +51,23 @@ exports.SignUp = function (req, res) {
     return res.status(500).json({ message: "server error" });
   }
 };
+
+exports.CheckCode = function (req, res) {
+  var { code, id } = req.body;
```
Comment (farshid0594) [inline src/controllers/app/userController.js]: chnage var to const

File: src/controllers/app/userController.js [hunk 2]
```diff
@@ -51,3 +51,23 @@ exports.SignUp = function (req, res) {
     return res.status(500).json({ message: "server error" });
   }
 };
+
+exports.CheckCode = function (req, res) {
+  var { code, id } = req.body;
+
+  User.findById(id)
+    .populate("code")
```
Comment (farshid0594) [inline src/controllers/app/userController.js]: Why populate with code?
Is The Code field reference to another model?

File: src/controllers/app/userController.js [hunk 3]
```diff
@@ -51,3 +51,23 @@ exports.SignUp = function (req, res) {
...
+  User.findById(id)
+    .populate("code")
+    .exec((err, foundedUser) => {
+      if (err) {
+        return res.status(500).json({ message: "server error" });
+      }
+      if (foundedUser === null) {
+        return res.status(404).json({ message: "user not found" });
+      }
+      if (foundedUser.code && foundedUser.code === code) {
```
Comment (farshid0594) [inline src/controllers/app/userController.js]: Where did you check the code expiration date?
foundedUser.code or foundedUser.code.text?

File: src/controllers/app/userController.js (hunk 1 above)
Comment (MohammadKoolivand) [reply src/controllers/app/userController.js]: done

File: src/controllers/app/userController.js [hunk 4]
```diff
@@ -1,4 +1,5 @@
 const moment = require("moment");
+var jwt = require("jsonwebtoken");
```
Comment (farshid0594) [inline src/controllers/app/userController.js]: change var to const

File: src/controllers/app/userController.js [hunk 5]
```diff
@@ -53,21 +54,29 @@ exports.SignUp = function (req, res) {
...
-    .exec((err, foundedUser) => {
-      if (err) {
-        return res.status(500).json({ message: "server error" });
-      }
-      if (foundedUser === null) {
-        return res.status(404).json({ message: "user not found" });
-      }
-      if (foundedUser.code && foundedUser.code === code) {
-        return res.status(200).json({ message: "succecfully confirmed" });
+  User.findById({ _id: id }, (err, foundedUser) => {
```
Comment (farshid0594) [inline src/controllers/app/userController.js]: User.findById({ _id: id } 
change to
User.findById(id,

File: src/controllers/app/userController.js [hunk 6]
```diff
@@ -53,21 +54,29 @@ exports.SignUp = function (req, res) {
...
+    if (err) {
+      return res.status(500).json({ message: "server error" });
+    }
+    if (foundedUser === null) {
+      return res.status(404).json({ message: "user not found" });
+    } else {
+      if (foundedUser.code.text === code) {
```
Comment (farshid0594) [inline src/controllers/app/userController.js:81]: Where did you check expiration of code?

File: src/controllers/app/userController.js [hunk 7]
```diff
@@ -53,21 +54,29 @@ exports.SignUp = function (req, res) {
...
+        const operationId = jwt.sign(
+          { userId: foundedUser._id, code: code },
+          process.env.JWT_SECRET_KEY,
+          {
+            expiresIn: process.env.JWT_EXPIRES_TIME, // 2h
```
Comment (farshid0594) [inline src/controllers/app/userController.js]: delete JWT_EXPIRES_TIME in .env file

Comment (MohammadKoolivand) [reply src/controllers/app/userController.js]: done

File: src/controllers/app/userController.js (hunk 5 above)
Comment (MohammadKoolivand) [reply src/controllers/app/userController.js]: done

**Review by farshid0594 — CHANGES_REQUESTED** [review]
Please resolve reviews

File: src/controllers/app/userController.js (hunk 4 above)
Comment (MohammadKoolivand) [reply src/controllers/app/userController.js]: done

File: src/controllers/app/userController.js [hunk 8]
```diff
@@ -56,19 +56,34 @@ exports.SignUp = function (req, res) {
...
+        moment().diff(
+          moment(foundedUser.code.expired, "YYYYMMDDThhmmss"),
+          "s"
+        ) > 0
+      ) {
+        return res
+          .status(400)
+          .json({
+            message: moment().diff(
+              moment(foundedUser.code.expired, "YYYYMMDDThhmmss"),
```
Comment (farshid0594) [inline src/controllers/app/userController.js:76]: change message Dr.koolivand.

**Review by farshid0594 — DISMISSED** [review]
Good job

**Review by alikia1995 — APPROVED** [review]

**Review by alikia1995 — APPROVED** [review]

**Review by alikia1995 — APPROVED** [review]

**Review by alikia1995 — APPROVED** [review]

**Review by alikia1995 — APPROVED** [review]

**Review by alikia1995 — APPROVED** [review]

**Review by alikia1995 — APPROVED** [review]

**Review by alikia1995 — APPROVED** [review]

**Review by alikia1995 — APPROVED** [review]

**Review by amirhnr — APPROVED** [review]

**Pull request merged**